lib/webauthn/attestation.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require "cbor"
+require "openssl"
+require "webauthn/attestation_statement"
+require "webauthn/authenticator_data"
+
+module WebAuthn
+  class Attestation
+    def self.deserialize(attestation_object)
+      from_map(CBOR.decode(attestation_object))
+    end
+
+    def self.from_map(map)
+      new(
+        authenticator_data: WebAuthn::AuthenticatorData.new(map["authData"]),
+        attestation_statement: WebAuthn::AttestationStatement.from(map["fmt"], map["attStmt"])
+      )
+    end
+
+    attr_reader :authenticator_data, :attestation_statement
+
+    def initialize(authenticator_data:, attestation_statement:)
+      @authenticator_data = authenticator_data
+      @attestation_statement = attestation_statement
+    end
+
+    def valid_attested_credential?
+      authenticator_data.attested_credential_data_included? &&
+        authenticator_data.attested_credential_data.valid?
+    end
+
+    def valid_attestation_statement?(client_data_hash)
+      @type, @trust_path = attestation_statement.valid?(authenticator_data, client_data_hash)
+    end
+
+    def trustworthy?
+      case type
+      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_NONE
+        acceptable_types.include?('None')
+      when WebAuthn::AttestationStatement::ATTESTATION_TYPE_SELF
+        acceptable_types.include?('Self')
+      else
+        acceptable_types.include?(type) && root_store.verify(subject_certificate, intermediate_certificates)
+      end
+    end
+
+    def aaguid
+      authenticator_data.aaguid
+    end
+
+    def certificate_key
+      attestation_statement.certificate_key
```

Comment:
I was eyeing to rename this and append `_id`, I think it's potentially confusing with a public/private key.

Reply:
Makes sense :+1:

Reply:
Done